### ISSUE TITLE
array/map to string cast behavior

### ISSUE THREAD

**Issue by tgpfeiffer** [sensorbee/sensorbee#58]
The documentation states:

> **2.4.3.4. To string**
> ...
> **From array**
> An array value is formatted as a JSON array:
>
>     [1, "2", 3.4]::string -- => "[1,""2"",3.4]"
> **From map**
> A map value is formatted as a JSON object:
>
>     {"a": 1, "b": "2", "c": 3.4}::string -- => "{""a"":1,""b"":""2"",""c"":3.4}"

I am not sure if this is a bug in the documentation, in the code, or in the shell, but when `EVAL`ing these examples in the shell, I get the following:

    > EVAL [1, "2", 3.4]::string;
    "[1,\"2\",3.4]"
    > EVAL {"a": 1, "b": "2", "c": 3.4}::string;
    "{\"a\":1,\"b\":\"2\",\"c\":3.4}"

These results work very well when thrown into the JavaScript `eval()` function, but they are not valid BQL strings. Is this actually a shell issue?

**Comment by tgpfeiffer** [comment]
OK, I got it. `{"a": 1, "b": "2", "c": 3.4}::string` results in the string

    `{"a": 1, "b": "2", "c": 3.4}`  // Go syntax

and the SensorBee shell prints this in JSON-encoded form. In fact, running

    > EVAL "{""a"":1,""b"":""2"",""c"":3.4}";
    "{\"a\":1,\"b\":\"2\",\"c\":3.4}"

gives the same result as evaluating the map-to-string code. I assume this is intended behavior, will add a note about that to the documentation and close this issue. If anyone thinks we should change the behavior of the shell, please reopen this issue.

**Comment by nobu-k** [comment]
No, it isn't an intended behavior. The result of `EVAL "{""a"":1,""b"":""2"",""c"":3.4}";` should be `"{""a"":1,""b"":""2"",""c"":3.4}"`, not `"{\"a\":1,\"b\":\"2\",\"c\":3.4}"`. Results of statements should be printed in the valid BQL syntax. Results of other statements like `SELECT` should be printed in the same way.

**Comment by nobu-k** [comment]
It might also be necessary to provide a way to print results in a JSON compatbile format.

**Comment by nobu-k** [comment]
This problem only affects the shell, so I'll provide a BQL compatbile formatting for now. It shouldn't be hard to switch JSON/BQL format.
